src/Microsoft.TestPlatform.ObjectModel/TestObject.cs

```diff
@@ -224,7 +239,18 @@ public void SetPropertyValue<T>(TestProperty property, LazyPropertyValue<T> valu
 
             object objValue = ConvertPropertyFrom<T>(property, culture, value);
 
-            this.PrivateSetPropertyValue(property, objValue);
+            this.PrivateSetPropertyValue(property, objValue, this.store);
+        }
+
+        /// <summary>
+        /// Set TestProperty's value to the specified value T.
+        /// </summary>
+        protected void SetLocalPropertyValue<T>(TestProperty property, T value)
```

Comment:
I am not sure if this is the right thing to do. In protocol V2 too, user will unnecessarily pay the cost of an additional dictionary and hashing several properties in it.